opal-ui/src/components/admin/providers/AddIdentityProviderDialog.vue

```diff
@@ -0,0 +1,348 @@
+                  <template v-slot:default>
+                    <span
+                      class="text-secondary"
+                      v-html="$t('identity_provider.use_logout', { url: useLogoutDefinition })"
+                    ></span>
+                  </template>
+                </q-checkbox>
+                <q-input
+                  v-model.number="newProvider.connectTimeout"
+                  dense
+                  type="number"
+                  :label="$t('identity_provider.connect_timeout')"
+                  :hint="$t('identity_provider.connect_timeout_hint')"
+                  class="q-mb-md"
+                  lazy-rules
+                >
+                </q-input>
+                <q-input
+                  v-model.number="newProvider.readTimeout"
+                  dense
+                  type="number"
+                  :label="$t('identity_provider.read_timeout')"
+                  :hint="$t('identity_provider.read_timeout_hint')"
+                  class="q-mb-md"
+                  lazy-rules
+                >
+                </q-input>
+                <q-input
+                  v-model="newProvider.callbackURL"
+                  dense
+                  type="text"
+                  :label="$t('identity_provider.callback_url')"
+                  :hint="$t('identity_provider.callback_url_hint')"
+                  class="q-mb-md"
+                  lazy-rules
+                  :rules="[validateUri('validation.identity_provider.callback_url_format', false)]"
+                >
+                </q-input>
+              </q-expansion-item>
+            </q-list>
+          </div>
+        </q-form>
+      </q-card-section>
+
+      <q-separator />
+
+      <q-card-actions align="right" class="bg-grey-3"
+        ><q-btn flat :label="$t('cancel')" color="secondary" v-close-popup />
+        <q-btn flat :label="submitCaption" type="submit" color="primary" @click="onAddProvider" />
+      </q-card-actions>
+    </q-card>
+  </q-dialog>
+</template>
+
+<script lang="ts">
+export default defineComponent({
+  name: 'AddIdentityProviderDialog',
+});
+</script>
+<script setup lang="ts">
+import { IDProviderDto } from 'src/models/Opal';
+import { notifyError } from 'src/utils/notify';
+
+interface DialogProps {
+  modelValue: boolean;
+  provider: IDProviderDto | null;
+}
+
+const { t } = useI18n();
+const identityProvidersStore = useIdentityProvidersStore();
+const formRef = ref();
+const props = defineProps<DialogProps>();
+const emit = defineEmits(['update:modelValue']);
+const showDialog = ref(props.modelValue);
+const emptyProvider = {
+  name: '',
+  clientId: '',
+  secret: '',
+  discoveryURI: '',
+  scope: 'openid',
+  useNonce: true,
+  useLogout: true,
+  parameters: [],
+  enabled: false,
+  connectTimeout: 0,
+  readTimeout: 0,
+} as IDProviderDto;
+
+const groupsMappingOptionsIndices = {
+  groupsClaim: 0,
+  groupsScript: 1,
+};
+const groupsMappingOptions = ref([
```

Comment:
Don't need to be a ref, it will not change

Reply:
:+1: